src/_guides/language/effective-dart/design.md

```diff
@@ -1533,36 +1533,42 @@ The new syntax is a little more verbose, but is consistent with other locations
 where you must use the new syntax.
 
 
-### DO annotate with `Object` instead of `dynamic` to indicate any object is allowed.
+### AVOID using `dynamic` unless you want dynamic dispatch.
```

Comment:
How about, more simply:

> ### AVOID using `dynamic`

---

I'm not sure that "dynamic dispatch" is appropriate here. By nature of Dart being OO, dynamic dispatching happens "all the time" for all types; it isn't specific to `dynamic`.

If you really want an extra qualifier for the title how about:

> ### AVOID using `dynamic` unless you want to disable static checks

(or "... static checking")

Reply:
> I'm not sure that "dynamic dispatch" is appropriate here. By nature of Dart being OO, dynamic dispatching happens "all the time" for all types; it isn't specific to `dynamic`.

This is one of those corners where I wish we could more easily rely on everyone have the same shared lingo. I personally make a distinction between *dynamic* and *virtual* dispatch. The former means looking up a member entirely by name with no reference to any static type the value may or may not have. It's what you get in JS, Ruby, Python, etc. The latter allows polymorphism and methods to be overridden, but still relies on the static type of the value to determine whether a method call is allowed.

But this is probably something I can't rely on readers understanding. I like your suggestion. Done.

Reply:
> This is one of those corners where I wish we could more easily rely on everyone have the same shared lingo. 

I agree.

> I personally make a distinction between dynamic and virtual dispatch. 

Nice distinction.